odoo/addons/base/models/ir_actions_report.py

```diff
@@ -517,10 +517,28 @@ def barcode(self, barcode_type, value, width=600, height=100, humanreadable=0, q
             if barcode_type == 'QR' and quiet:
                 bar_border = 0
 
-            barcode = createBarcodeDrawing(
-                barcode_type, value=value, format='png', width=width, height=height,
-                humanReadable=humanreadable, quiet=quiet, barBorder=bar_border
-            )
+            kwargs = {
+                'value': value,
+                'format': 'png',
+                'width': width,
+                'height': height,
+                'humanReadable': humanreadable,
+                'quiet': quiet,
+                'barBorder': bar_border,
+            }
```

Comment:
Seems completely unnecessary and to just complexify the diff. Why not just *add* an optional kwargs dict, which can be set up (or not) by individual code types?

Reply:
Mhm, sorry, I can't actually understand what you mean with "which can be setup by individual code types". Can you elaborate, please?

Reply:
Currently the call is:
```python
barcode = createBarcodeDrawing(
                barcode_type, value=value, format='png', width=width, height=height,
                humanReadable=humanreadable, quiet=quiet, barBorder=bar_border
            )
```
It could just become
```python
barcode = createBarcodeDrawing(
                barcode_type, value=value, format='png', width=width, height=height,
                humanReadable=humanreadable, quiet=quiet, barBorder=bar_border,
                **kwargs
            )
```
and outside of the `try` block, `kwargs` would be defined to either an empty dict or one with options relevant to the barcode type. I guess some of the existing options could be folded into the kwargs (e.g. `barBorder` which is barcode-dependent to a degree) but updating the entire call doesn't seem necessary given most of it is in no way changed.